app/src/main/java/com/bogdan/codeforceswatcher/features/news/NewsFragment.kt

```diff
@@ -49,61 +115,112 @@ class NewsFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener, StoreSubs
     }
 
     override fun onNewState(state: NewsState) {
-        if (state.status == NewsState.Status.PENDING) {
-            swipeRefreshLayout.isRefreshing = true
-        } else {
-            swipeRefreshLayout.isRefreshing = false
-            val items = mutableListOf<NewsItem>()
-
-            val feedbackController = FeedbackController.get()
-
-            if (feedbackController.shouldShowFeedbackCell()) {
-                items.add(NewsItem.FeedbackItem(feedbackController.feedUIModel))
-                newsAdapter.callback = {
-                    onNewState(state)
-                }
-            }
+        val items = mutableListOf<NewsItem>()
 
-            val news = state.news.filter {
-                return@filter if (it is News.PinnedPost) settings.readLastPinnedPostLink() != it.link else true
-            }
+        val feedbackController = FeedbackController.get()
+        var feedbackCallback: () -> Unit = {}
 
-            val newsItems = buildNewsItems(news)
-            items.addAll(newsItems)
+        if (feedbackController.shouldShowFeedbackCell()) {
+            items.add(NewsItem.FeedbackItem(feedbackController.feedUIModel))
+            feedbackCallback = { onNewState(state) }
```

Comment:
Why don't we pass `feedbackCallback` directly in `ContentView`?

Reply:
Since it needs to pass last state, I can do it only here.

Reply:
If you change the state to use `NewsState` directly, you will have in `ContentView`.

Reply:
Do you suggest pass NewsState, FeedbackItem and callback as separate variables instead as one state?

Reply:
Or just use `NewsState` in `UIModel` directly and add there `FeedbackItem` instead add it in `NewsItems` list.

Reply:
Yes, the second suggestion.